System/Vector.cpp

```diff
@@ -53,32 +53,14 @@ namespace RTE {
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	float Vector::GetAbsRadAngle() const {
-		if (m_X == 0) {
-			return (m_Y > 0) ? -c_HalfPI : ((m_Y < 0) ? c_HalfPI : 0);
-		}
-		if (m_Y == 0) {
-			return (m_X > 0) ? 0 : ((m_X < 0) ? c_PI : 0);
-		}
-		// TODO: Confirm that this is correct!")
-		float rawAngle = -std::atan(m_Y / m_X);
-		if (m_X < 0) { rawAngle += c_PI; }
-
-		return rawAngle;
+		float radAngle = -std::atan2(m_Y,m_X);
```

Comment:
missing a space after the comma here
Also, it's been a while since I've thought about geometry and stuff, and though I'm pretty sure atan2 does get the angle we want, I don't understand the negative here. Is it to support +- angles?

Reply:
the minus is because angles in cc are in a regular coordinate system, i.e. pi/2 points up.
whereas the vectors are in an upside-down coordinate system, i.e. (0,-1) points up.

Reply:
Ahhh, right, that makes sense. Good stuff!